apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_enrollment_pre_surveyTest.jsx

```diff
@@ -0,0 +1,164 @@
+  });
+
+  describe("With Data", () => {
+    let userIndex = 0;
+    let fakeEnrollments;
+    let workshopEnrollmentPreSurvey;
+    let tableRows;
+    before(() => {
+      const generateFakeEnrollment = survey => {
+        userIndex++;
+        return {
+          first_name: `Teacher${userIndex}`,
+          last_name: `LastName${userIndex}`,
+          email: `teacher${userIndex}@ex.net`,
+          district_name: "A district",
+          school: "A school",
+          user_id: userIndex,
+          attended: true,
+          pre_workshop_survey: survey
+        };
+      };
+
+      fakeEnrollments = [
+        generateFakeEnrollment({
+          unit: "Unit 4 - the fouth unit",
+          lesson: "Lesson 3 - the third lesson",
+          questionsAndTopics: "so many questions",
+          unitLessonShortName: "U4 L3"
+        }),
+        generateFakeEnrollment({
+          unit: "Unit 1 - the first unit",
+          lesson: "Lesson 11 - the eleventh lesson",
+          questionsAndTopics: "another question",
+          unitLessonShortName: "U1 L11"
+        }),
+        generateFakeEnrollment({
+          unit: "Unit 1 - the first unit",
+          lesson: "Lesson 2 - the second lesson",
+          questionsAndTopics: "",
+          unitLessonShortName: "U1 L2"
+        }),
+        generateFakeEnrollment({
+          unit: "Unit 1 - the first unit",
+          lesson: "Lesson 1 - the first lesson",
+          questionsAndTopics: "",
+          unitLessonShortName: "U1 L1"
+        }),
+        generateFakeEnrollment({
+          unit: "Unit 4 - the fouth unit",
+          lesson: "Lesson 3 - the third lesson",
+          questionsAndTopics: "more questions...",
+          unitLessonShortName: "U4 L3"
+        }),
+        // one no response
+        generateFakeEnrollment(null)
+      ];
+      workshopEnrollmentPreSurvey = shallow(<WorkshopEnrollmentPreSurvey enrollments={fakeEnrollments}/>);
+      tableRows = workshopEnrollmentPreSurvey.find("Table tbody tr");
+    });
+
+    it("Has the expected table column headers", () => {
+      const columnHeaders = workshopEnrollmentPreSurvey.find("Table thead tr th").map(h => h.text());
+      expect(columnHeaders).to.eql([
+        "#",
+        "First Name",
+        "Last Name",
+        "Email",
+        "Current Unit",
+        "Current Lesson",
+        "Questions and topics they hope to discuss"
+      ]);
+    });
+
+    it("Displays one table row for each enrollment", () => {
+      expect(tableRows).to.have.length(6);
+    });
+
+    it("Displays survey responses", () => {
+      const ROW_FULL_RESPONSE = 0;
+      const responseCellText = tableRows.at(ROW_FULL_RESPONSE).find("td").map(td => td.text());
+      expect(responseCellText).to.eql([
+        `${ROW_FULL_RESPONSE + 1}`,
+        fakeEnrollments[ROW_FULL_RESPONSE].first_name,
+        fakeEnrollments[ROW_FULL_RESPONSE].last_name,
+        fakeEnrollments[ROW_FULL_RESPONSE].email,
+        "Unit 4 - the fouth unit",
```

Comment:
is 'fouth' a reference i don't understand? sorry if these were all intentional and i'm just missing the joke!

Reply:
Copy/paste error. Not intentional. Though now I want to come up with something... ;)

Reply:
Or rather, a successful copy/paste from the original typo